src/basic/fd-util.c

```diff
@@ -880,3 +880,20 @@ const char *format_fd_path(int fd, char **buf) {
          * empty string. */
         return path_equal(*buf, "/") ? "" : *buf;
 }
+
+int dir_fd_is_root(int dir_fd) {
+        _cleanup_close_ int rfd = -EBADF;
+        struct stat st, rst;
+
+        assert(dir_fd >= 0);
+
+        rfd = openat(dir_fd, "/", O_CLOEXEC|O_NOFOLLOW|O_PATH|O_DIRECTORY);
```

Comment:
Not that this matters greatly, but why use `dir_fd` here? It'll be ignored, but I imaging that looking at a strace, I would look at the first argument first, and wonder if the second one is a mistake…

Hmm, and why not just do `lstat("/", &rst)`? It's one syscall instead of three.